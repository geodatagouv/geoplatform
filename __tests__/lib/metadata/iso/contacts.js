const {getAllContacts} = require('../../../../lib/metadata/iso/contacts')

describe('lib.metadata.iso.contacts', () => {
  describe('getAllContacts()', () => {
    it('should return contacts array when called with representative metadata', () => {
      const metadata = {
        contact: {
          organisationName: 'Tralala',
          individualName: 'Régis',
          role: 'OWNER',
          contactInfo: {
            address: {
              deliveryPoint: '1 rue République',
              postalCode: '75015',
              city: 'Paris 15e',
              electronicMailAddress: 'info@acme.org',
              country: 'Groland'
            },
            phone: {
              voice: '0147200001',
              facSimile: '3615'
            }
          }
        },
        identificationInfo: {
          pointOfContact: [
            {
              organisationName: 'ACME'
            },
            {
              individualName: 'AAA'
            }
          ]
        }
      }

      expect(getAllContacts(metadata)).toMatchSnapshot()
    })
  })
})
