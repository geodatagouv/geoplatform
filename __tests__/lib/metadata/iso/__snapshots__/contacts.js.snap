// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.metadata.iso.contacts getAllContacts() should return contacts array when called with representative metadata 1`] = `
Array [
  Object {
    "address": "1 rue RÃ©publique",
    "country": "Groland",
    "email": "info@acme.org",
    "organizationName": "Tralala",
    "phoneNumber": "0147200001",
    "postalCode": "75015",
    "relatedTo": "metadata",
    "role": "owner",
    "town": "Paris 15e",
  },
  Object {
    "organizationName": "ACME",
    "relatedTo": "data",
    "role": "notDefined",
  },
]
`;
